File: commands/frankjpeg.py
import bot
import discord
import os


def import_command():

    # Create Command Group
    frank_jpeg_commands = discord.app_commands.Group(
        name='frankjpeg',
        description='Choose a specific Frank jpeg'
    )

    # Add Command Group to Tree
    bot.tree.add_command(frank_jpeg_commands)

    # Number Command
    @frank_jpeg_commands.command(
        name="num",
        description="Choose a Frank jpeg by number"
    )
    async def send_frank_jpeg_num(interaction: discord.Interaction, option: int):
        await interaction.response.send_message(file = discord.File(bot.assets[option]))

    # Name Command
    @frank_jpeg_commands.command(
        name="name",
        description="Choose a Frank jpeg by name, including the file extension"
    )
    async def send_frank_jpeg_name(interaction: discord.Interaction, option: str):
        # Get just the file names
        jpegNames = [os.path.basename(asset).casefold() for asset in bot.assets]
        jpegIndex = jpegNames.index(option.casefold())
        await interaction.response.send_message(file = discord.File(bot.assets[jpegIndex]))

    # List Sub-Command
    @frank_jpeg_commands.command(
        name='list',
        description='List Frank jpegs'
    )
    async def list_frank_jpegs(interaction: discord.Interaction):
        message = 'Frank jpegs:'
        for asset_id, jpegName in enumerate(bot.assets):
            message += f'\n*{asset_id}*: **{os.path.basename(jpegName)}**'
        await interaction.response.send_message(message)

File: src/commands/chat/frankjpeg.js
const path = require("path");
const { assets } = require("../../index.js");
const { SlashCommandBuilder } = require("discord.js");

module.exports = {
  data: new SlashCommandBuilder()
    .setName("frankjpeg")
    .setDescription("Choose a specific Frank jpeg")
    .addSubcommand((subcommand) =>
      subcommand
        .setName("num")
        .setDescription("Choose a Frank jpeg by number")
        .addNumberOption((option) =>
          option
            .setName("number")
            .setRequired(true)
            .setDescription(
              "Type in the number based on entries in /frankjpeg list"
            )
        )
    )
    .addSubcommand((subcommand) =>
      subcommand
        .setName("name")
        .setDescription("Choose a Frank jpeg by name")
        .addStringOption((option) =>
          option
            .setName("name")
            .setRequired(true)
            .setDescription(
              "Type the name you found in /frankjpeg list (including extension)"
            )
        )
    )
    .addSubcommand((subcommand) =>
      subcommand.setName("list").setDescription("List Frank jpegs")
    ),

  async num(ctx, num) {
    let picture = path.join(__dirname, "../../../assets/images", assets[num]);
    await ctx.reply({ files: [{ attachment: picture }] });
  },

  async name(ctx, name) {
    let picture = path.join(__dirname, "../../../assets/images", name);
    await ctx.reply({ files: [{ attachment: picture }] });
  },

  async list(ctx) {
    let message = "```\nFrank Jpegs:";
    let number = 0;
    for (let asset in assets) {
      message += `\n${number}. ${assets[asset]}`;
      number += 1;
    }
    message += "\n```";
    await ctx.reply(message);
  },

  async execute(ctx) {
    let subcommand = ctx.options.getSubcommand();
    if (subcommand === "num") {
      await this.num(ctx, ctx.options.getNumber("number"));
    } else if (subcommand === "name") {
      await this.name(ctx, ctx.options.getString("name"));
    } else if (subcommand === "list") {
      await this.list(ctx);
    }
  },
};
